src/config/pragmas/validate/_lib.js

```diff
@@ -0,0 +1,44 @@
+let is = require('../../../lib/is')
+
+let patterns = {
+  looseName: new RegExp(/^[a-z][a-zA-Z0-9-_]+$/),
+  strictName: new RegExp(/^[a-z][a-z0-9-]+$/),
+  // DynamoDB, SNS, SQS
+  veryLooseName: new RegExp(/^[a-zA-Z0-9/\-._]*$/),
+}
+
+function regex (value, pattern, pragmaName, errors) {
+  if (!patterns[pattern]) throw ReferenceError(`Invalid validation pattern specified: ${pattern}`)
```

Comment:
How come some of the errors in this module's methods `throw`, and some push onto the `errors` parameter? I'm not understanding how the difference is used - do you mind elaborating into this a bit for me?

Example: `validate/_events` executes this module's `unique` method right away, and there's a chance that that throws, which wouldn't leverage the new `errors` param being passed around. Am I understanding this right?

Reply:
Aye, good question! Any of the `validate/_lib` stuff that throws would be for internal programmer errors to these internal methods. To your example: if `unique` ever got !array, that's on us, not the user, and we should explode loudly. I could probably have written a bunch more conditions to throw in those little methods but I just wanted to protect the gnarlier ones – arguable we don't need them though, as unit tests should (in theory) catch misuse-related issues. wdyt?

Reply:
Got it, so `throw` is used to eject out ASAP, whereas `errors` is collected in order to provide one or more (likely?) user (app author) errors back to the user (app authors) for correction?

Reply:
Yep!

Reply:
Cool thanks!